source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -1031,28 +980,17 @@ namespace
             else if (typeName == RtNode::typeName())
             {
                 NodePtr mxNode = writeNode(prim, doc);
-
-                if (writeOptions)
+                if (mxNode->getCategory() == SURFACE_MATERIAL_NODE_STRING && writeOptions &&
```

Comment:
To be more general you should be able to do something like:
`mxNode->getNodeDef()->getType() == MATERIAL_TYPE_STRING`